src/Factory/TermFactory.php

```diff
@@ -60,9 +60,16 @@ protected function from_id(int $id): ?Term
         return $this->build($wp_term);
     }
 
-    protected function from_wp_term_query(WP_Term_Query $query): iterable
+    protected function from_wp_term_query(WP_Term_Query $query) 
```

Comment:
@nlemoine Would the function not always return an `iterable` result, even if it’s an array of ids?

Reply:
It might as well return a string (if `'fields' == 'count'`), which is not an iterable. So we need to remove it.

Reply:
This is a tricky situation.

Should `Timber::get_terms()` exactly match `get_terms` behavior?

Since we can't use union types yet (`int|iterable`), I would say: `Timber::get_terms()` should be used for getting objects or ids. If you need count, use `get_terms`. There's actually no benefit in using `Timber::get_terms()` for getting anything else than objects (event IDs) and keeping return types seems more important to me.

Reply:
@nlemoine - I see your point. My fix does more than is needed to fix the bug, and probably than is reasonable. 

On the other hand - if I pass a `WP_Term_Query` object to `TermFactory::from()` (even indirect via `Timber::get_terms()`) - what would you expect to happen? If the query is about the number of terms (i.e., `fields == count`) we should return exactly that. 
Or, to phrase it differently, what should we do if it is set like that and we actually need to return an `iterable`: throw an error? return an empty array?

Also note the method actually calling this protected, typed method is `TimberFactory::from()` - which does not have a return type and also simply ignores the return types of all methods it is calling (just passing on the results).

Reply:
You're right @jrathert, the problem lies elsewhere. Maybe in the fact that we allow a `WP_Term_Query` to be passed or maybe we should throw something here, after all, it's a **term** factory, not a count term factory :) You should expect an iterable here.

Let's drop the return type then.

Reply:
The problem lies within the weird behavior/return types of `WP_Term_Query::get_terms()` I'd say. I am a bit new to PHP and WordPress, but I find it astonishing how pragmatic the language can be used - and actually is within WordPress.

Reply:
PHP ≠ WordPress :) WordPress kind of lives outside regular PHP world.